flagsmith/models.py

```
@@ -29,15 +29,18 @@ class Flag(BaseFlag):
     def from_evaluation_result(
         cls,
         flag_result: SDKFlagResult,
-    ) -> typing.Optional[Flag]:
+    ) -> Flag:
         if metadata := flag_result.get("metadata"):
             return Flag(
                 enabled=flag_result["enabled"],
                 value=flag_result["value"],
                 feature_name=flag_result["name"],
                 feature_id=metadata["flagsmith_id"],
             )
-        return None
+        raise ValueError(
+            "FlagResult metadata is missing. Cannot create Flag instance. "
+            "This means a bug in the SDK, please report it."
+        )
```

Sorry, I'm probably missing some context, but [this](https://github.com/Flagsmith/flagsmith-engine/blob/main/flag_engine/result/types.py#L20) indicates to me that metadata isn't required, so why should we throw an error here?

---

See the thread [here](https://github.com/Flagsmith/flagsmith-python-client/pull/163/files/9b23e749ec504eaa8777401aa8368cd495c966b2#r2437097773).

---

Thanks for the pointer. Makes sense now.